include/beast/http/file_body_linux.hpp

```diff
@@ -0,0 +1,256 @@
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BEAST_HTTP_FILE_BODY_LINUX_HPP
+#define BEAST_HTTP_FILE_BODY_LINUX_HPP
+
+#ifdef __linux__
+# define BEAST_NO_LINUX_FILE 0
+#else
+# define BEAST_NO_LINUX_FILE 1
+#endif
+
+#if ! BEAST_NO_LINUX_FILE
+
+#include <beast/core/async_result.hpp>
+#include <beast/core/error.hpp>
+#include <beast/core/file_base.hpp>
+#include <beast/http/message.hpp>
+#include <beast/http/serializer.hpp>
+#include <boost/asio/buffer.hpp>
+#include <boost/asio/basic_stream_socket.hpp>
+#include <boost/filesystem.hpp>
+#include <csignal>
+#include <sys/sendfile.h>
+#include <unistd.h>
+
+namespace beast {
+namespace http {
+
+struct file_body_linux
+{
+    class reader;
+    class writer;
+
+    class value_type
+    {
+        int file_{-1};
+        std::uint64_t size_{0};
+
+        friend class reader;
+        friend class writer;
+
+        template<int Signal>
+        static void signal_init()
+        {
+            static auto x{std::signal(Signal, SIG_IGN)};
+            (void)x;
+        }
+
+    public:
+        value_type(const value_type&) = delete;
+        value_type& operator=(const value_type&) = delete;
+
+        /** Destructor.
+
+            If the file handle is open, it is closed first.
+        */
+        ~value_type()
+        {
+            if(file_ != -1)
+                ::close(file_);
+        }
+
+        /// Default constructor
+        value_type() = default;
+
+        /// Move constructor
+        value_type(value_type&& other) noexcept
+            : file_(other.file_)
+            , size_(other.size_)
+        {
+            other.file_ = -1;
+            other.size_ = 0;
+        }
+
+        /// Move assignment
+        value_type& operator=(value_type&& other) noexcept
+        {
+            value_type tmp(std::move(other));
+            std::swap(file_, tmp.file_);
+            std::swap(size_, tmp.size_);
+            return *this;
+        }
+
+        /// Returns the native file handle
+        int
+        native_handle() const
+        {
+            return file_;
+        }
+
+        /// Returns `true` if the file is open
+        bool
+        is_open() const
+        {
+            return file_ != -1;
+        }
+
+        /** Open a file for reading or writing.
+
+            @param path The path to the file
+
+            @param mode The open mode used with fopen()
+
+            @param ec Set to the error, if any occurred
+        */
+        void
+        open(boost::filesystem::path const& path,
+            file_mode mode, error_code& ec)
+        {
+            int f;
+            switch(mode)
+            {
+            default:
+            case file_mode::read:   f = O_RDONLY; break;
+            case file_mode::scan:   f = O_RDONLY; break;
+            case file_mode::write:  f = O_WRONLY | O_CREAT; break;
+            case file_mode::append: f = O_WRONLY | O_APPEND; break;
```

Comment:
You mentioned elsewhere that you had another implementation of posix-based files you can use, so I assume you'll merge that before this becomes a real feature. I only planned to implement just enough here to support the prototype.